sdk/eventhub/Azure.Messaging.EventHubs/src/Core/TransportProducerPool.cs

```diff
@@ -0,0 +1,360 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Azure.Core;
+
+namespace Azure.Messaging.EventHubs.Core
+{
+    /// <summary>
+    ///   A pool of <see cref="TransportProducer" /> instances that automatically expire after a period of inactivity.
+    /// </summary>
+    ///
+    internal class TransportProducerPool : IAsyncDisposable
+    {
+        /// <summary>The period in minutes after which <see cref="PerformExpiration" /> is run.</summary>
+        private static readonly TimeSpan DefaultPerformExpirationPeriod = TimeSpan.FromMinutes(10);
+
+        /// <summary>
+        ///   The set of active Event Hub transport-specific producers specific to a given partition;
+        ///   intended to perform delegated operations.
+        /// </summary>
+        ///
+        private ConcurrentDictionary<string, PoolItem> Pool { get; }
+
+        /// <summary>
+        ///   An abstracted Event Hub transport-specific producer that is associated with the
+        ///   Event Hub gateway rather than a specific partition; intended to perform delegated operations.
+        /// </summary>
+        ///
+        public TransportProducer EventHubProducer { get; }
+
+        /// <summary>
+        ///   A reference to a <see cref="Timer" /> periodically checking every <see cref="DefaultPerformExpirationPeriod" />
+        ///   the <see cref="TransportProducer" /> that are in use and those that can be closed.
+        /// </summary>
+        ///
+        private Timer ExpirationTimer { get; }
+
+        /// <summary>
+        ///   Initializes a new instance of the <see cref="TransportProducerPool" /> class.
+        /// </summary>
+        ///
+        /// <param name="eventHubProducer">An abstracted Event Hub transport-specific producer that is associated with the Event Hub gateway rather than a specific partition.</param>
+        /// <param name="pool">The pool of <see cref="PoolItem" /> that is going to be used to store the partition specific <see cref="TransportProducer" />.</param>
+        /// <param name="performExpirationPeriod">The period after which <see cref="PerformExpiration" /> is run. Overrides <see cref="DefaultPerformExpirationPeriod" />.</param>
+        ///
+        public TransportProducerPool(TransportProducer eventHubProducer,
+                                     ConcurrentDictionary<string, PoolItem> pool = default,
+                                     TimeSpan? performExpirationPeriod = default)
+        {
+            performExpirationPeriod ??= DefaultPerformExpirationPeriod;
```

Comment:
(extreme) nit:  Would be nice to have these in the same order as the parameters.